app/controllers/profiles_controller.rb

```diff
@@ -0,0 +1,21 @@
+class ProfilesController < ApplicationController
+  before_action :set_user
```

Comment:
it should be applied only for show action
`before_action :set_profile, only: :show`